server/mqtt.go

```diff
@@ -1466,20 +1478,21 @@ func (jsa *mqttJSA) prefixDomain(subject string) string {
 
 func (jsa *mqttJSA) newRequestEx(kind, subject, cidHash string, hdr int, msg []byte, timeout time.Duration) (interface{}, error) {
 	var sb strings.Builder
-	jsa.mu.Lock()
 	// Either we use nuid.Next() which uses a global lock, or our own nuid object, but
 	// then it needs to be "write" protected. This approach will reduce across account
 	// contention since we won't use the global nuid's lock.
+	jsa.mu.Lock()
```

Comment:
Nit, not related to the fix. Doesn't seem that anything other than getting the uuid needs to be under the lock.

Reply:
Technically there is `jsa.rplyr` but this is immutable, so should be fine done outside.